app/javascript/components/rules/index.js

```diff
@@ -1,67 +1,102 @@
+    if (!rule.enabled)
+      status = <Badge status="new">Disabled</Badge>;
+    else if (rule.events.length === 0) {
+      status = <Badge status="new">No activity</Badge>;
+    }
+    else if (rule.events.length > 0 && rule.events[rule.events.length - 1].error) {
+      status = <Badge status="warning">Warning</Badge>;
+    }
+    else {
+      status = <Badge status="success">Healthy</Badge>;
+    }
+
     return (
-      <AppProvider>
-        <Layout>
-          <Layout.AnnotatedSection
-            id="ruleIndex"
-            title="Rules"
-            description="These are the rules you created."
-          >
+      <Card>
+        <Card.Section title={ rule.name }
+          actions={[
             {
-              this.props.rules.map(function(rule, index) {
-                let badge;
+              content: 'Activity',
+              url: `/rules/${ rule.id }/activity`
+            },
+            {
+              content: 'Edit',
+              url: `/rules/${ rule.id }`
+            },
+            {
+              content: 'Delete',
+              url: `/rules/${ rule.id }`,
+              onAction: (event) => {
+                if (event.currentTarget.getAttribute('data-remote'))
+                  return;
+
+                event.preventDefault();
+                event.stopPropagation();
 
-                if (rule.enabled) {
-                  badge = <Badge status="success">Enabled</Badge>
-                } else {
-                  badge = <Badge status="warning">Disabled</Badge>
+                if (window.confirm('Delete this rule?')) {
+                  event.currentTarget.setAttribute("data-remote", "remote");
+                  event.currentTarget.setAttribute("data-method", "delete");
+
+                  event.currentTarget.click();
                 }
-                
-                return (
-                  <Card key={ index }>
-                    <Card.Section title={ rule.name }
-                      actions={[
-                        {
-                          content: 'Activity',
-                          url: `/rules/${ rule.id }/activity`
-                        },
-                        {
-                          content: 'Edit',
-                          url: `/rules/${ rule.id }`
-                        },
-                        {
-                          content: 'Delete',
-                          url: `/rules/${ rule.id }`,
-                          onAction: (event) => {
-                            if (event.currentTarget.getAttribute('data-remote'))
-                              return;
+              }
+            }
+          ]}
+        >
+          <p>
+            { status }
+          </p>
+        </Card.Section>
+      </Card>
+    );
+  }
+}
+
+class Index extends React.Component {
+  render () {
+    let rules_enabled = this.props.rules.filter(function(x) { return x.enabled });
+    let rules_disabled = this.props.rules.filter(function(x) { return !x.enabled });
```

Comment:
I was curious to know if there was the equivalent to Ruby's array.partition, in order to avoid looping over the array twice, and turns out there isn't. But! People tend to use alternatives that I find kinda nice: https://codereview.stackexchange.com/questions/162878/partitioning-an-array-based-on-a-condition-in-javascript